### PR TITLE
Qsv default device

### PR DESCRIPTION
**Pull request by xhaihao**

### CODE REVIEW

File: libavutil/hwcontext_d3d11va.c [hunk 1]
```diff
@@ -581,10 +631,24 @@ static int d3d11va_device_create(AVHWDeviceContext *ctx, const char *device,
...
+    } else {
+        AVDictionaryEntry *e = av_dict_get(opts, "vendor_id", NULL, 0);
+        if (e && e->value) {
+            adapter = d3d11va_device_find_adapter_by_vendor_id(ctx, creationFlags, e->value);
+            if (adapter < 0) {
+                av_log(ctx, AV_LOG_ERROR, "Failed to find d3d11va adapter by "
+                       "vendor id %s\n", e->value);
+                return AVERROR_UNKNOWN;
+            } else
+                av_log(ctx, AV_LOG_VERBOSE, "Selecting adapter index %d\n", adapter);
```
Comment (galinart) [inline libavutil/hwcontext_d3d11va.c]: I would reword this to "Selecting d3d11va adapter %d\n" and move this line to 649 for both use cases

File: libavutil/hwcontext_d3d11va.c [hunk 2]
```diff
@@ -581,10 +631,24 @@ static int d3d11va_device_create(AVHWDeviceContext *ctx, const char *device,
...
+        }
+    }
+
+    if (adapter >= 0) {
         IDXGIFactory2 *pDXGIFactory;
```
Comment (galinart) [inline libavutil/hwcontext_d3d11va.c:648]: Put the av_log here

**Comment by galinart** [comment]
Tested on my NVIDIA + ADL configuration and works fine.

File: libavutil/hwcontext_d3d11va.c (hunk 1 above)
Comment (xhaihao) [reply libavutil/hwcontext_d3d11va.c]: Updated, thx

**Comment by xhaihao** [comment]
https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=10129
